course/src/isometry.cc

```diff
@@ -307,6 +312,110 @@ const Matrix3 Matrix3::kZero =
 const Matrix3 Matrix3::kOnes = 
     Matrix3(Vector3(1, 1, 1), Vector3(1, 1, 1), Vector3(1, 1, 1));
 
+// Isometry class methods
+
+Isometry Isometry::FromTranslation(const Vector3& v) {
+    return Isometry(v);
+}
+
+Isometry Isometry::RotateAround(const Vector3& v, const double rotation_angle) {
+
+    const double cangle{std::cos(rotation_angle)};
+    const double sangle{std::sin(rotation_angle)};
+
+    Matrix3 rot_matrix;
+
+    if (v == Vector3::kUnitX) {
+        rot_matrix = Matrix3(Vector3::kUnitX,
+                             Vector3(0, cangle, -sangle),
+                             Vector3(0, sangle, cangle));
+    }
+    else if (v == Vector3::kUnitY) {
+        rot_matrix = Matrix3(Vector3(cangle, 0, sangle),
+                             Vector3::kUnitY,
+                             Vector3(-sangle, 0, cangle));
+    }
+    else if (v == Vector3::kUnitZ) {
+        rot_matrix = Matrix3(Vector3(cangle, -sangle, 0),
+                             Vector3(sangle, cangle, 0),
+                             Vector3::kUnitZ);
+    }
+    else {
+        throw "Invalid rotation axis";
+    }
+
+    return Isometry(Vector3(), rot_matrix);
+
+
+}
+
+Isometry Isometry::FromEulerAngles(const double roll, 
+                                   const double pitch,
+                                   const double yaw) {
+    return Isometry(Isometry::RotateAround(Vector3::kUnitX, roll) *
+                    Isometry::RotateAround(Vector3::kUnitY, pitch) *
+                    Isometry::RotateAround(Vector3::kUnitZ, yaw));
+}
+
+// Operators
+Matrix3& Isometry::operator = (const Matrix3& m) {
+    translation_vector_ = Vector3();
+    rotation_matrix_ = m;
+    return rotation_matrix_;
+}
+
+Vector3 Isometry::operator * (const Vector3& v) const {
+    return Vector3(rotation_matrix_ * (v + translation_vector_));
+}
+
+Isometry Isometry::operator * (const Isometry& t) const {
+    return Isometry(translation() + t.translation(),
+                    rotation() ^ t.rotation());
+}
+
+bool Isometry::operator == (const Isometry& t) const {
+    return (rotation() == t.rotation()) && 
+           (translation() == t.translation());
+}
+
+// Misc
+
+Vector3 Isometry::transform(const Vector3& v) const {
+    return (*this) * v;
+}
+
+Isometry Isometry::compose(const Isometry& t) const {
+    Vector3 trans(get_x_translation() + t.get_x_translation(),
+                  get_y_translation() + t.get_y_translation(),
+                  get_z_translation() + t.get_z_translation());
+    
+    return Isometry(trans, rotation_matrix_ ^ t.rotation_matrix_);
+}
+
+// Getters
+Isometry Isometry::inverse() const {
+    return Isometry(this->translation() * -1);
```

Comment:
This the inverse. The inverse of an isometry must be such that a vector transformed by the product of the isometry by the inverse does not change.

Reply:
Ok, now it should be done well!